x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/promql/PromqlParserUtils.java

```diff
@@ -272,4 +274,23 @@ private static int adjustColumn(int lineNumber, int columnNumber, int startColum
         // the column offset only applies to the first line of the PROMQL command
         return lineNumber == 1 ? columnNumber + startColumn - 1 : columnNumber;
     }
+
+    /*
+     * Parses a Prometheus date which can be either a float representing epoch seconds or an RFC3339 date string.
+     */
+    public static Instant parseDate(Source source, String value) {
+        try {
+            return Instant.ofEpochMilli((long) (Double.parseDouble(value) * 1000));
+        } catch (NumberFormatException ignore) {
+            // Not a float, try parsing as date string
+        }
+        if (value.startsWith("\"") && value.endsWith("\"")) {
+            value = value.substring(1, value.length() - 1);
+        }
```

Comment:
See if AbstractBuilder#unquote applies here (since ESQL supports also raw / unescaped strings using triple quotes: https://www.elastic.co/docs/reference/query-languages/esql/esql-syntax#esql-string-literals)